### ISSUE TITLE
Add new formatter "constraint"

### ISSUE THREAD

**Issue by warpy** [ansman/validate.js#157]
Hi there,

I propose adding a new formatter called "constraints". My request is to have validate.js return ONLY the constraint names as the validation result. Ex:

```
var c = {
  foo: {
    numericality: true,
    length: {
      is: 23
    }
  }
};
var o = {
  format: 'constraint'
};
var errors = validate({ foo: "bar" }, c, o));

/*
errors = {
  foo: ["length", "numericality"]
}
*/
```

This is really useful because I can then create localized error messages in my UI based on the constraint that fails.I think doing this help solve #69 as well, so people don't have to rely on the message formatted by validate.js and can instead render their own error message.

I have the code change ready to merge this. If there's already a way to do it, then please let me know and I can close the issue. I know we can use `{ message: '^numericality' }` to get a similar result but this essentially makes it much easier and less verbose.

Thanks!

**Comment by huan086** [comment]
My "hack" to achieve the custom error message is to change `convertErrorMessages` to return the keypath of the `constraints` object where validation fail. So instead of

```js
errors = {
  foo: ["length", "numericality"]
}
```
I have
```
errors = {
  foo: ["length.is", "numericality"]
}
```

This way, I'm able to tell whether to display "{field} must be less than {count} characters", "{field} must be more than {count} characters" or "{field} must be {count} characters"

**Comment by ansman** [comment]
A fix has been released in 0.12.0
